src/Compilers/CSharp/Test/Semantic/Semantics/OutVarTests.cs

```diff
@@ -14591,5 +14575,199 @@ static int Test1(out int x)
+  {
+    ret
+  }
+  // i = 1; return 2;
+  .method public virtual instance int32 get_P([out] int32& i)
+  {
+    ldarg.1
+    ldc.i4.1
+    stind.i4
+    ldc.i4.2
+    ret
+  }
+  // i = 3; return;
+  .method public virtual instance void set_P([out] int32& i, int32 v)
+  {
+    ldarg.1
+    ldc.i4.3
+    stind.i4
+    ret
+  }
+  .property instance int32 P([out] int32&)
+  {
+    .get instance int32 A::get_P([out] int32&)
+    .set instance void A::set_P([out] int32&, int32)
+  }
+  // i = 4; return 5;
+  .method public virtual instance int32 get_Item([out] int32& i)
+  {
+    ldarg.1
+    ldc.i4.4
+    stind.i4
+    ldc.i4.5
+    ret
+  }
+  // i = 6; return;
+  .method public virtual instance void set_Item([out] int32& i, int32 v)
+  {
+    ldarg.1
+    ldc.i4.6
+    stind.i4
+    ret
+  }
+  .property instance int32 Item([out] int32&)
+  {
+    .get instance int32 A::get_Item([out] int32&)
+    .set instance void A::set_Item([out] int32&, int32)
+  }
+}";
+            var reference1 = CompileIL(source1);
+            var source2 =
+@"using System;
+class B
+{
+    public static void Main()
+    {
+        A a = new A();
+        IA ia = a;
+        Console.WriteLine(ia.P[out int i1] + "" "" + i1);
+        ia.P[out int i2] = 4;
+        Console.WriteLine(i2);
+        Console.WriteLine(ia[out int i3] + "" "" + i3);
+        ia[out int i4] = 4;
+        Console.WriteLine(i4);
+    }
+}";
+            var compilation2 = CompileAndVerify(source2, additionalRefs: new[] { reference1 }, expectedOutput:
+@"2 1
+3
+5 4
+6");
+            compilation2.VerifyIL("B.Main()",
+@"{
+  // Code size      103 (0x67)
+  .maxstack  4
+  .locals init (int V_0, //i1
+                int V_1, //i2
+                int V_2, //i3
+                int V_3) //i4
+  IL_0000:  newobj     ""A..ctor()""
+  IL_0005:  dup
+  IL_0006:  ldloca.s   V_0
+  IL_0008:  callvirt   ""int IA.P[out int].get""
+  IL_000d:  box        ""int""
+  IL_0012:  ldstr      "" ""
+  IL_0017:  ldloc.0
+  IL_0018:  box        ""int""
+  IL_001d:  call       ""string string.Concat(object, object, object)""
+  IL_0022:  call       ""void System.Console.WriteLine(string)""
+  IL_0027:  dup
+  IL_0028:  ldloca.s   V_1
+  IL_002a:  ldc.i4.4
+  IL_002b:  callvirt   ""void IA.P[out int].set""
+  IL_0030:  ldloc.1
+  IL_0031:  call       ""void System.Console.WriteLine(int)""
+  IL_0036:  dup
+  IL_0037:  ldloca.s   V_2
+  IL_0039:  callvirt   ""int IA.this[out int].get""
+  IL_003e:  box        ""int""
+  IL_0043:  ldstr      "" ""
+  IL_0048:  ldloc.2
+  IL_0049:  box        ""int""
+  IL_004e:  call       ""string string.Concat(object, object, object)""
+  IL_0053:  call       ""void System.Console.WriteLine(string)""
+  IL_0058:  ldloca.s   V_3
+  IL_005a:  ldc.i4.4
+  IL_005b:  callvirt   ""void IA.this[out int].set""
+  IL_0060:  ldloc.3
+  IL_0061:  call       ""void System.Console.WriteLine(int)""
+  IL_0066:  ret
+}");
+        }
     }
```

Comment:
Are you saying that code in ParseVariableDeclarator is not reachable? Otherwise please add a test that hits that code path.

---
In reply to: [75945475](https://github.com/dotnet/roslyn/pull/13274#discussion_r75945475) [](ancestors = 75945475,75905658,75798494,75795201,75786568,75782994)